### ISSUE TITLE
vim.bo.undolevels wrong value, most likely underflow? 

### ISSUE THREAD

**Issue by ms-jpq** [neovim/neovim#15587]
### Neovim version (nvim -v)

v0.6.0-dev

### Vim (not Nvim) behaves the same?

N.A.

### Operating system/version

ubuntu / mac

### Terminal name/version

iTerm2

### $TERM environment variable

xterm-256color

### Installation

ubuntu: unstable repo, mac: homebrew stable

### How to reproduce the issue

1. open nvim

2. `:edit` a file

3. `lua print(vim.bo.undolevels)`

### Expected behavior

the number printed should be either the global undo level or according to docs:

```txt
	The local value is set to -123456 when the global value is to be used.
```

### Actual behavior

a very large number is printed, different depending on either macos or ubuntu.

I suspect this is a case of underflow happening.

---

I discovered this when trying to set `vim.bo.undolevels = vim.bo.undolevels` to break undo sequences

**Comment by gpanders** [comment]
This is a result of the return value being an uninitialized number, which is addressed in #15592. However, there is still a larger issue of a mismatch with how local options are handled in Lua and VimL (#14670).

**Comment by gpanders** [comment]
Closed by #15996
